geomstats/learning/natural_gradient_descent.py

```diff
@@ -0,0 +1,120 @@
+"""Natural gradient descent using Amari's anaytic solution.
+
+References
+----------
+.. [AS1985] Amari S., Karakida R., Oizumi M. (2018)
+    Fisher Information and Natural Gradient Learning of Random Deep Networks.
+"""
+
+import math
+
+import numpy as np
+import torch
+from torch.optim import Optimizer
+
+import geomstats.backend as gs
+
+
+class NaturalGradientDescent(Optimizer):
+    """Natural Gradient Descent.
+
+    Uses Amari's analytic solution for the inverse Fisher information matrix.
+
+    Parameters
+    ----------
+    params : array-like, shape=[3, dim]
+        The current layer of the ResNet, consisting of a vector of nodes,
+        a vector of weights, and a bias scalar
+    lr : float
+        The learning rate to update the weights
+    wd : float
+        Penalization on weights to prevent overfitting
+    bias : Boolean
+        Whether or not to include bias term in training
+
+    References
+    ----------
+    .. [AS1985] Amari S., Karakida R., Oizumi M. (2018)
+        Fisher Information and Natural Gradient Learning of Random Deep Networks.
+    """
+
+    def __init__(self, params, lr=0.05, wd=0.0, bias=True):
+        defaults = dict(lr=lr, wd=wd, bias=bias)
+        self.params = params
+
+        super().__init__(params, defaults)
+
+    def step(self):
+        """Perform one step of Natural Gradient Descent."""
+        w_list = []
+        w_star_list = []
+        w_tilde_list = []
+        for group in self.param_groups:  # only 1 group for now
```

Comment:
what does this comment mean?

Reply:
there is by default only one parameter group, unless you choose to add more. I can delete the comment